### PR TITLE
Fixes #30987 - Add card template extension to host details page

### PR DESCRIPTION
**Pull request by amirfefer**
This PR adds the ability to add a card to the host details overview tab via plugins.

- [x] determine Card structure with header and content
- [x] Create responsive cards layout (the number of cards is determined by the screen size)
- [x] Adding a slot component for a multiple extension in the  overview tab
- [x] Adding Card Item Template component with a key-value data-driven approach

### CODE REVIEW

**Comment by amirfefer** [comment]
can be tested with https://github.com/Katello/katello/pull/9011

File: webpack/assets/javascripts/react_app/components/HostDetails/Parameters/index.js [hunk 1]
```diff
@@ -27,67 +30,71 @@ const ParametersCard = ({ paramters }) => {
...
   return (
     <Card isHoverable>
-      <CardBody>{__('Parameters')}</CardBody>
-      <Accordion asDefinitionList>
-        {!paramters.length && (
-          <div style={{ marginLeft: '20px' }}>
-            <Skeleton count={3} width={200} />
-          </div>
-        )}
+      <CardTitle component="h4">{__('Paramterers')}</CardTitle>
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/HostDetails/Parameters/index.js]:
```suggestion
      <CardTitle component="h4">{__('Parameterers')}</CardTitle>
```

File: webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import CardItem from './CardItem';
+
+import { addGlobalFill } from '../../common/Fill/GlobalFill';
+
+const CardTemplate = ({ content, header, slotID, cardID }) => {
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js]: We are not using the CardTemplate anywhere. Is it just for reference? If so, I think we should move it to some different folder and use it in a StoryBook?

File: webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js [hunk 3]
```diff
@@ -2,6 +2,8 @@ import PropTypes from 'prop-types';
 import React, { useEffect, useState } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 import Skeleton from 'react-loading-skeleton';
+import ArrowIcon from '@patternfly/react-icons/dist/js/icons/arrow-icon';
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: Is it wise to import from internals of `@patternfly`? Aren't they exporting it somehow?

File: webpack/assets/javascripts/react_app/components/HostDetails/Parameters/styles.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.pf-c-accordion {
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/HostDetails/Parameters/styles.scss]: Should we namespace it a bit? or do we really want to apply the rule on all the accordions?

**Review by ezr-ondrej — COMMENTED** [review]
In general make sense and the `CardItem`/`CardBody` looks much cleaner! :+1: 
Left few thoughts inline, let me know what you think.

**Comment by ezr-ondrej** [comment]
+ the snapshots needs refresh.

File: webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js (hunk 2 above)
Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js]: we use this`cardItem` component in plugins, i.e - https://github.com/theforeman/foreman_remote_execution/pull/549

File: webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js (hunk 3 above)
Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: This import is taken by pf4 docs - https://github.com/patternfly/patternfly-react/tree/master/packages/react-icons

File: webpack/assets/javascripts/react_app/components/common/CardTemplate/CardItem.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+      setActiveAccordion(id);
+    }
+  };
+
+  return (
+    <GridItem xl2={3} md={6} lg={5}>
+      <Card isHoverable>
+        <CardTitle>{header}</CardTitle>
+        <CardBody>
+          {children || (
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/CardTemplate/CardItem.js]: This enables us to add custom content from plugins.
You can find a usage example here https://github.com/theforeman/foreman_remote_execution/pull/549

File: webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js (hunk 2 above)
Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js]: Nope, I see the `CardItem` used there :thinking:

File: webpack/assets/javascripts/react_app/components/common/CardTemplate/CardItem.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  CardTitle,
+  CardBody,
+  Accordion,
+  AccordionItem,
+  AccordionContent,
+  AccordionToggle,
+  GridItem,
+} from '@patternfly/react-core';
+
+const CardItem = ({ content, header, children }) => {
```
Comment (ezr-ondrej) [inline webpack/assets/javascripts/react_app/components/common/CardTemplate/CardItem.js]: This seems to have pretty specific implementation details, do we want to have this for all Cards?
Shouldn't we just start with a `HostDetails/template/CardItem` and only once we tweek it and find another usage, let's generalize? Generalizing soon is usually bad idea.

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
I still don't see the `CardTemplate` to be used anywhere and I would prefer it to be a HostSpecific Component for now, until we know how we want to use cards on other places :thinking:
But maybe I'm missing something?

File: webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js (hunk 2 above)
Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/common/CardTemplate/index.js]: You're right, it's probably a leftover :)
deleted

**Comment by amirfefer** [comment]
@ezr-ondrej 
updated :)

**Review by ezr-ondrej — APPROVED** [review]
It would be nice to unify the Cards (reduce the boilerplate a bit) in the future, but this works nicely for now :+1:
Thanks @amirfefer !

File: webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js (hunk 3 above)
Comment (evgeni) [inline webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: This broke RPM builds:

```
ModuleNotFoundError: Module not found: Error: Can't resolve '@patternfly/react-icons/dist/js/icons/arrow-icon' in '/builddir/build/BUILD/foreman-2.4.0-develop/webpack/assets/javascripts/react_app/components/HostDetails/Audits'
```

Do we need a specific version for that to work?

Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: We do have rather an old version of this package in our packaging, but I can't find any clue on this package versioning and thus what version we need. @amirfefer ?

Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: The file `arrow-icon.js` exists even in an old version like `3.15.16`
We import patternfly 4 icons without any error
i.e
https://github.com/theforeman/foreman/blob/c4559edf0802a10ced3cdcee8620718e438383ba/webpack/assets/javascripts/react_app/components/Layout/components/CustomContextSelector.js#L12

Comment (evgeni) [reply webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: But never from dist. Is that an issue?

Comment (amirfefer) [reply webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: @evgeni  This import is based on patternfly usage suggestion (for tree shaking) - https://github.com/patternfly/patternfly-react/tree/master/packages/react-icons

Hopefully, this PR fixes the issue https://github.com/theforeman/foreman/pull/8190

Comment (evgeni) [reply webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js:5]: Yeah, not sure what makes it exactly fail here. The linked PR looks like Worth a try, thanks!

**Pull request merged**